docs/ecto.md

```diff
@@ -0,0 +1,447 @@
+
+With our files in place, let's follow the instructions and run our migration. If you forgot to run the `mix ecto.create` task after creating your application, do that now with `mix ecto.create`. Next run:
+
+```console
+$ mix ecto.migrate
+Compiling 1 file (.ex)
+Generated hello_phoenix app
+
+[info]  == Running HelloPhoenix.Repo.Migrations.CreateHelloPhoenix.User.change/0 forward
+
+[info]  create table users
+
+[info]  == Migrated in 0.0s
+```
+
+Mix assumes that we are in the development environment unless we tell it otherwise with `MIX_ENV=another_environment mix some_task`. Our Ecto task will get its environment from Mix, and that's how we get the correct suffix to our database name.
+
+If we log in to our database server, and connect to our `hello_phoenix_dev` database, we should see our `users` table. Ecto assumes that we want an integer column called `id` as our primary key, so we should see a sequence generated for that as well.
+
+```console
+$ psql -U postgres
+
+Type "help" for help.
+
+postgres=# \connect hello_phoenix_dev
+You are now connected to database "hello_phoenix_dev" as user "postgres".
+hello_phoenix_dev=# \d
+                List of relations
+ Schema |       Name        |   Type   |  Owner
+--------+-------------------+----------+----------
+ public | schema_migrations | table    | postgres
+ public | users             | table    | postgres
+ public | users_id_seq      | sequence | postgres
+(3 rows)
+hello_phoenix_dev=# \q
+```
+
+If we take a look at the migration generated by `phx.gen.schema` in `priv/repo/migrations`, we'll see that it will add the columns we specified. It will also add timestamp columns for `inserted_at` and `updated_at` which come from the `timestamps/0` function.
+
+```elixir
+defmodule HelloPhoenix.Repo.Migrations.CreateHelloPhoenix.User do
+  use Ecto.Migration
+
+  def change do
+    create table(:users) do
+      add :name, :string
+      add :email, :string
+      add :bio, :string
+      add :number_of_pets, :integer
+
+      timestamps()
+    end
+
+  end
+end
+```
+
+And here's what that translates to in the actual `users` table.
+
+```console
+hello_phoenix_dev=# \d users
+Table "public.users"
+Column     |            Type             | Modifiers
+----------------+-----------------------------+----------------------------------------------------
+id             | integer                     | not null default nextval('users_id_seq'::regclass)
+name           | character varying(255)      |
+email          | character varying(255)      |
+bio            | character varying(255)      |
+number_of_pets | integer                     |
+inserted_at    | timestamp without time zone | not null
+updated_at     | timestamp without time zone | not null
+Indexes:
+"users_pkey" PRIMARY KEY, btree (id)
+```
+
+Notice that we do get an `id` column as our primary key by default, even though it isn't listed as a field in our migration.
```

Comment:
Maybe mention how to change this in a migration? Or a link off to the Ecto docs?